### PR TITLE
added mypy-diff and mypy-commit targets to makefile

### PR DESCRIPTION
**Pull request by sanderr**
This commit attempts to simplify identifying newly introduced type errors by providing the mypy-diff make target. This target compares the output of running mypy with the current baseline. The baseline represents the type errors we currently condone. This diff allows for easy tracking of newly introduced or resolved type errors. The mypy-commit target updates the baseline file. Running this for each pull requests should give a clear overview of potential changes to the type coverage.
I am not entirely satisfied with the name "baseline", so suggestions on that are welcome.

### CODE REVIEW

**Comment by sanderr** [comment]
I could also use some feedback on the diff colors. I opted for red for newly introduced errors and green for resolved errors, since that reflects the good/bad intuition. However, this is the opposite of git file diffs, which might be counterintuitive.

**Comment by sanderr** [comment]
I added the line `mypy_test_var: int = 2.1` at the end of a file, resulting in the following diff:
mypy-diff

**Comment by sanderr** [comment]
I just realized I didn't take line numbers in mypy output into account. This means that adding or removing any lines in a file, results in a diff for all type errors below that line, since the line numbers have changed. I'll have to think on that before this is ready to be merged in.

**Comment by sanderr** [comment]
Bart and I were already discussing whether the file should be tracked or not. That's mainly why I added you as a reviewer, to get an extra opinion on this. Conclusion: it should not be tracked.
On another note, I've been thinking that it might be cleaner to use `# type: ignore` annotations instead. By which I mean we'd do a single commit that adds these annotations to all currently condoned errors. That would actually make this whole mypy-diff obsolete. I should have thought about this before starting on the tool.
@bartv, @wouterdb What's your opinion on the annotations vs the mypy-diff?

**Comment by bartv** [comment]
We do have the ambition to fix most of the type errors

**Comment by sanderr** [comment]
Of course. But either of these solutions would be a way to cope with the current type errors in the meantime. The question is: which is preferable?

**Comment by sanderr** [comment]
Conclusion after discussion in person: no type ignores.

**Comment by sanderr** [comment]
I updated the tool so it doesn't flag line changes as newly introduced errors. It's pretty hacky, but it's contained, and it works. I believe this will make the tool more useful.

File: Makefile [hunk 1]
```diff
@@ -25,9 +25,34 @@ pep8:
...
 
-.PHONY: mypy
+.PHONY: mypy mypy-diff mypy-commit
+RUN_MYPY=MYPYPATH=stubs:src python -m mypy --html-report mypy -p inmanta
+
 mypy:
-	MYPYPATH=stubs:src python -m mypy --html-report mypy -p inmanta
+	$(RUN_MYPY)
+
+MYPY_TMP_FILE=.mypy-tmp
```
Comment (wouterdb) [inline Makefile]: this whole magic could use some docs.

Comment (sanderr) [reply Makefile]: What exactly do you mean? Docs on how to use it? Some comments in the makefile explaining the magic, ...?

Comment (sanderr) [reply Makefile]: I've pushed some changes regarding this comment.

**Pull request merged**